specification/cognitiveservices/Language.Conversations/common.tsp

```diff
@@ -0,0 +1,694 @@
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+}>;
+
+@doc("job state")
+@added(Versions.v2023_04_01)
+model JobState {
+  @doc("display name")
+  displayName?: string;
+
+  @doc("Date and time job created")
+  createdDateTime: utcDateTime;
+
+  @doc("Date and time job expires")
+  expirationDateTime?: utcDateTime;
+
+  @doc("job ID")
+  @key
+  @visibility("read")
+  jobId: uuid;
+
+  @doc("last updated date and time")
+  lastUpdatedDateTime: utcDateTime;
+
+  @doc("status")
+  status: State;
+
+  @doc("errors")
+  errors?: Array<Error>;
+
+  @doc("next link")
+  nextLink?: string;
+}
+
+@doc("job errors")
+model JobErrors {
+  @doc("errors")
+  errors: Array<Error>;
+}
+
+@doc("Contains details of errors encountered during a job execution.")
+model DocumentError {
+  @doc("The ID of the input document.")
+  id: string;
+
+  @doc("Error encountered.")
+  error: Error;
+}
+
+@doc("Contains details of warnings encountered during a job execution.")
+model InputWarning {
+  @doc("Warning code.")
+  code: string;
+
+  @doc("Warning message.")
+  message: string;
+
+  @doc("A JSON pointer reference indicating the target object.")
+  targetRef?: string;
+}
+
+@doc("if showStats=true was specified in the request this field will contain information about the request payload.")
+model RequestStatistics {
+  @doc("Number of documents submitted in the request.")
+  documentsCount: int32;
+
+  @doc("Number of valid documents. This excludes empty, over-size limit or non-supported languages documents.")
+  validDocumentsCount: int32;
+
+  @doc("Number of invalid documents. This includes empty, over-size limit or non-supported languages documents.")
+  erroneousDocumentsCount: int32;
+
+  @doc("Number of transactions for the request.")
+  transactionsCount: int64;
+}
+
+@doc("Pre built result")
+model PreBuiltResult {
+  @doc("Errors by document id.")
+  errors: Array<DocumentError>;
+
+  @doc("statistics")
+  statistics?: RequestStatistics;
+
+  @doc("This field indicates which model is used for scoring.")
+  modelVersion: string;
+}
+
+@doc("Represents List of Question Answers.")
+model AnswersResult {
+  @doc("Represents Answer Result list.")
+  answers?: Array<KnowledgeBaseAnswer>;
+}
+
+@doc("Represents knowledge base answer.")
+model KnowledgeBaseAnswer {
+  @doc("List of questions associated with the answer.")
+  questions?: Array<string>;
+
+  @doc("Answer text.")
+  answer?: string;
+
+  @doc("Answer confidence score, value ranges from 0 to 1.")
+  @minValue(0)
+  @maxValue(1)
+  @clientName("confidence")
+  confidenceScore?: float64;
+
+  @doc("ID of the QnA result.")
+  @clientName("qnaId")
+  id?: int32;
+
+  @doc("Source of QnA result.")
+  source?: string;
+
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "KnowledgeBaseAnswer.metadata should not use ': Record<unknown>'. Use ': Record<string>' instead."
```

Comment:
Let's remove the suppression if it no longer applies.

Reply:
@quentinRobinson I am not familiar with why we used the Record<unknown> type for QnA response. Are you aware of this and can we change this to Record<string>?

Reply:
`Record<string>` is a string keyed map/dictionary. `Record<unknown>` would be like `additionalProperties`.

Reply:
It can be removed since it moved to `Record<unkown>`

Reply:
done.